docs/multi-agent-applications.md

```diff
@@ -0,0 +1,333 @@
+from pydantic_ai_examples.sql_gen import system_prompt
+
+# Multi-agent Applications
+
+There are roughly four levels of complexity when building applications with PydanticAI:
+
+1. Single agent workflows — what most of this documentation covers
+2. [Agent delegation](#agent-delegation) — agents using another agent via tools
+3. [Programmatic agent hand-off](#programmatic-agent-hand-off) — one agent runs, then application code calls another agent
+4. [Graph based control flow](#pydanticai-graphs) — for the most complex cases, graph and a state machine can be used to control the execution of multiple agents
+
+Of course, you can combine multiple strategies in a single application.
+
+## Agent Delegation
+
+The agent delegates work to another agent, but then takes back control when that agent finishes.
```

Comment:
```suggestion
"Agent delegation" refers to the scenario where an agent delegates work to another agent, but then takes back control when that agent finishes.
```

Reply:
Instead of "that agent", maybe we could say something that clarifies which agent we mean